storage/src/main/java/org/apache/kafka/storage/internals/log/UnifiedLog.java

```diff
@@ -1887,8 +1887,10 @@ public int deleteOldSegments() throws IOException {
             return deleteLogStartOffsetBreachedSegments() +
                     deleteRetentionSizeBreachedSegments() +
                     deleteRetentionMsBreachedSegments();
-        } else {
+        } else if (config().compact) {
             return deleteLogStartOffsetBreachedSegments();
+        } else {
+            return deleteLogStartOffsetBreachedSegments() + deleteRetentionSizeBreachedSegments();
```

Comment:
This is not quite right. If `cleanup.policy` is empty, we want to ignore `log.retention.bytes` and `log.retention.ms` and only pick up `log.local.retention.bytes` and `log.local.retention.ms` if remote storage is enabled.

Reply:
Well, an empty cleanup.policy is the same as delete with retention time and retention size set to -1, which tiered storage supports. So, empty cleanup.policy should be supported by tiered storage. In the KIP, we have the following.

`If cleanup.policy is empty and remote.storage.enable is set to true, the local log segments will be cleaned based on the values of log.local.retention.bytes and log.local.retention.ms.`

Reply:
That means `validateRemoteStorageRequiresDeleteCleanupPolicy` should be removed, since an empty policy is supported by remote storage now. Also, `deleteOldSegments` should handle all deletions when `remoteLogEnabled` is true.

Reply:
We still need `validateRemoteStorageRequiresDeleteCleanupPolicy()`, but we need to adjust it to support empty cleanup policy.

Reply:
> Well, an empty cleanup.policy is the same as delete with retention time and retention size set to -1, which tiered storage supports. So, empty cleanup.policy should be supported by tiered storage. In the KIP, we have the following.

Thanks for the explaination!